File: src/main/java/com/uapp_llc/repository/TaskRepository.java
package com.uapp_llc.repository;

import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.uapp_llc.model.Task;

public interface TaskRepository {

  Task save(Task entity);

  Optional<Task> findByIdAndColumnId(Long id, Long columnId);

  Page<Task> findAllByColumnId(Long columnId, Pageable pageable);

  void delete(Task entity);

}

File: src/main/kotlin/com/uapp_llc/repository/TaskRepository.kt
package com.uapp_llc.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import com.uapp_llc.model.Task

interface TaskRepository {

  fun save(entity: Task): Task

  fun findByIdAndColumnId(id: Long, columnId: Long): Task?

  fun findAllByColumnId(columnId: Long, pageable: Pageable): Page<Task>

  fun delete(entity: Task)

}
